src/realm/sync/client.cpp

```diff
@@ -1338,14 +1316,19 @@ void SessionWrapper::async_wait_for(bool upload_completion, bool download_comple
     REALM_ASSERT(upload_completion || download_completion);
 
     m_client.post([self = util::bind_ptr{this}, handler = std::move(handler), upload_completion,
-                   download_completion]() mutable {
+                   download_completion](Status status) mutable {
         REALM_ASSERT(self->m_actualized);
+        if (!status.is_ok()) {
```

Comment:
If `post()` itself failed we previously never called the completion callback, while now we report the error to the callback. The event loop being able to fail is sort of weird and I'm not sure it can actually happen?